src/components/ProgressAnimation/index.js

```diff
@@ -0,0 +1,148 @@
+          }}
+        />
+      ) : (
+        ''
+      )}
+      <svg
+        xmlns="http://www.w3.org/2000/svg"
+        xmlnsXlink="http://www.w3.org/1999/xlink"
+        width="32"
+        height="32"
+        className={`${namespace}__indicator`}
+      >
+        <defs />
+        <defs>
+          <circle
+            id={`circle--${progressAnimationName}`}
+            cx="16"
+            cy="16"
+            r="16"
+          />
+          <mask
+            id={`mask--${progressAnimationName}`}
+            width="32"
+            height="32"
+            x="0"
+            y="0"
+            fill="#fff"
+            maskContentUnits="userSpaceOnUse"
+            maskUnits="objectBoundingBox"
+          >
+            <use xlinkHref={`#circle--${progressAnimationName}`} />
+          </mask>
+        </defs>
+        <g fill="none" fillRule="evenodd">
+          <circle cx="16" cy="16" r="15" stroke={filledColor} strokeWidth="2" />
+          <use
+            stroke={unfilledColor}
+            strokeWidth="4"
+            mask={`url(#mask--${progressAnimationName})`}
+            transform="matrix(1 0 0 -1 0 32)"
+            xlinkHref={`#circle--${progressAnimationName}`}
+            style={
+              // If the component is animated:
+              animationTimer
+                ? {
+                    strokeDasharray: 100,
+                    strokeDashoffset: 0,
+                    animationName: progressAnimationName,
+                    animationDuration: `${animationTimer}s`,
+                    animationTimingFunction: 'linear',
+                    animationFillMode: 'forwards',
+                  }
+                : // If the component is NOT animated:
+                  {
+                    strokeDasharray: 100,
+                    strokeDashoffset: percentage,
+                  }
+            }
+          />
+        </g>
+      </svg>
+      {children && <div className={`${namespace}__message`}>{children}</div>}
+    </div>
+  );
+}
+
+ProgressAnimation.propTypes = {
+  /** Children contains the message displayed next to the animation. */
+  children: PropTypes.node,
+
+  /** Custom class name applied to component wrapper. */
+  className: PropTypes.string,
+
+  /** The percentage (number only) that represents progress measured, out of 100. */
+  percentage: PropTypes.number.isRequired,
+
+  /** The filled color of the progress animation svg, representing percentage complete. */
+  filledColor: PropTypes.string,
+
+  /** the unfilled color of the progress animation svg, representing percetnage incomplete. */
+  unfilledColor: PropTypes.string,
+
+  /**
+   * Optional custom timer for this component instance's animation, in seconds.
+   * If value is nonexistant (i.e., set to its default `null`, `0`, or `undefined`),
+   * then the component will not be animated at all.
+   */
+  animationTimer: PropTypes.number,
+
+  /** Unique id for this component instance. */
+  id: PropTypes.number.isRequired,
```

Comment:
- [ ] I recommend enforcing this as a string here as it will provide better flexibility for `id`s that most likely would be generated with both alphabetical and numerical characters

Reply:
I thought about that, however this ID is appended to a class name so i worried that allowing a string would open it up to non-classname-friendly characters. 🤔

Reply:
I said class name above but i meant the name of the keyframe animation 😅